com/course/zenkouv/src/Lecture3/Task1/Knight.java

```diff
@@ -0,0 +1,124 @@
+package Lecture3.Task1;
+
+import Lecture3.Task1.comparator.CostComparator;
+import Lecture3.Task1.comparator.WeightComparator;
+import Lecture3.Task1.model.Armor;
+import Lecture3.Task1.model.Outfit;
+import Lecture3.Task1.model.Weapon;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class Knight {
+
+    private final String name;
+    private final List<Outfit> outfitList;
+    private int outfitsCost;
+
+    public Knight(String name) {
+        this.name = name;
+        this.outfitList = new ArrayList<>();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public List<Outfit> getOutfitList() {
+        return outfitList;
+    }
+
+    public int getOutfitsCost() {
+        return this.outfitsCost;
+    }
+
+    public void addOutfitToOutfitLis(Outfit outfit) {
+        if (!availabilityOutfit(outfit)) {
+            outfitList.add(outfit);
+        }
+    }
+
+    private boolean availabilityOutfit(Outfit outfit) {
+        if (isWeapon(outfit)) {
```

Comment:
if you add new type of Outfit it will be harder to expend this method
use switch case

Reply:
I don't understand how to use it in this case.

Reply:
`  private` boolean outfitChecker(Outfit outfit) {
        switch (outfit.getClass().getSimpleName()) {
            case "Weapon": {
                for (Outfit outfitFromList : outfits) {
                    if (outfitFromList instanceof Weapon) {
                        System.out.printf("I have a %s and me don't need any weapon!\n",
                                ((Weapon) outfitFromList).getType().toString().toLowerCase());
                        return true;
                    }
                }
            }
            case "Armor": {
                for (Outfit outfitFromList : outfits) {
                    if (outfitFromList instanceof Armor) {
                        if (((Armor) outfit).getType() == ((Armor) outfitFromList).getType()) {
                            System.out.printf("I have a %s and me don't need any %s!\n",
                                    ((Armor) outfitFromList).getType().toString().toLowerCase(),
                                    ((Armor) outfitFromList).getType().toString().toLowerCase());
                            return true;
                        }
                    }
                }
            }
        }
        return false;
    } `